server/src/main/java/keywhiz/service/resources/automation/v2/SecretResource.java

```diff
@@ -326,7 +326,7 @@ public Iterable<String> secretListing(@Auth AutomationClient automationClient,
   public Iterable<SanitizedSecret> secretListingV2(@Auth AutomationClient automationClient,
       @QueryParam("idx") Integer idx, @QueryParam("num") Integer num,
       @DefaultValue("true") @QueryParam("newestFirst") boolean newestFirst) {
-    permissionCheck.checkAllowedOrThrow(automationClient, Action.READ);
+    permissionCheck.checkAllowedForTargetTypeOrThrow(automationClient, Action.READ, SanitizedSecret.class);
 
```

Comment:
Our code makes a distinction between SanitizedSecret and regular ol' Secret, so I decided that the check should as well (even if the implementation just coalesces everything into Secret, although this code wouldn't know/shouldn't assume that).

Also, note that phase 1 doesn't support field-level permissions (but phase 2 will), so for cases where we're just returning a single field such as secret name we just check permissions for the entire Secret class, which could be overly restrictive.